rs/replicated_state/src/canister_state/queues/queue.rs

```diff
@@ -483,28 +448,27 @@ impl TryFrom<(&CanisterQueue, &MessagePool)> for OutputQueue {
         let mut output_queue = OutputQueue::new(q.capacity);
         let mut request_slots = 0;
         let mut response_slots = 0;
-        for queue_item in q.iter() {
-            let id = queue_item.id();
-            let msg = match pool.get(id) {
+        for reference in q.iter() {
+            let msg = match pool.get(*reference) {
                 Some(msg) => msg.clone(),
 
                 // Stale request, skip it.
-                None if id.kind() == Kind::Request => {
+                None if reference.kind() == Kind::Request => {
```

Comment:
Similar here, but I guess less critical because this code is only supposed to run on up/downgrade.

Reply:
Indeed, this only runs on downgrade (and there's similar code on upgrade). I wrote it like this because it would be quite hard to reproduce the stale entries. And pointless, since all replicas are restarting at once.

Reply:
Makes sense. Thank you.